src/standalone_custom/shore_power_randomize.py

```diff
@@ -1,51 +1,147 @@
-total_shore_power = pd.Series([0 for i in range(0, len(index))], index=index)
-
-for column in shore_power.columns:
-    correct_combination = False
-    while correct_combination != True:
-        print("Get shore power times for: ", column, " docking time")
-        time_of_events = pd.Series(
-            [shore_power[column]["Power"] for i in range(0, len(index))], index=index
-        ).sample(
-            n=shore_power[column]["Number"], replace=False  # number of events
-        )
-        correct_combination = True
-        number = 0
-        for i in time_of_events.index:
-            for j in time_of_events.index:
-                if i != j:
-                    if j in pd.date_range(
-                        start=i,
-                        end=i
-                        + pd.DateOffset(hours=int(shore_power[column]["Duration"])),
-                        freq="h",
-                    ):
-                        correct_combination == False
-                    else:
-                        for hour in range(0, shore_power[column]["Duration"]):
-                            if total_shore_power[i + pd.DateOffset(hours=hour)] == 0:
-                                total_shore_power[
-                                    i + pd.DateOffset(hours=hour)
-                                ] = shore_power[column]["Power"]
-                            else:
-                                correct_combination == False
-
-            number += 1
-            print(
-                "Valid random event ",
-                number,
-                " of total ",
-                shore_power[column]["Number"],
+logging.basicConfig(level=logging.DEBUG)
+
+
+def randomize_shore_power(
+    times, shore_power, filename="randomized_shore_power_demand.csv", docks=1
+):
+    r"""
+    Creates randomized shore power time series.
+
+    times : pandas.DatetimeIndex
+        Time period for which randomized shore power time series is created.
+    shore_power : pandas.DataFrame
+        Specification of shore power with row indices 'Duration' for the
+        duration of a docking period of a ship, 'Number' showing the amount of
+        docking events during `times` and 'Power` specifying the power of the
+        respective ship. Add one column for each group of ships that share
+        the same power and docking duration (name free to choose). It is
+        assumed that ships of one column cannot occupy a dock at the same time,
+        ships of different columns can though, if `docks` is greater than one.
+    filename : str
+        File name including path for saving the shore power time series.
+        Default: "demand_shore_power.csv".
+    docks : int
+        Amount of docks that are available simultaneously. Default: 1.
+
+    Returns
+    -------
+    total_shore_power : pandas.Series
+        Randomized shore power time series.
+
+    Todo
+    -----
+    * Adapt format of `total_shore_power` for MVS. (f.e. header, index ...)
```

Comment:
Does this mean the file can currently not be directly used for MVS? (Okay, just need to know).

Reply:
I haven't checked that. We need to define what the file "is allowed" to look like - as we were discussing before.
But you're right - would be handy to at least implement a solution that is working now. I'll check that but would still keep the ToDo.

Reply:
Done.